src/calendar.cpp

```diff
@@ -398,6 +398,14 @@ int calendar::season_length()
     return cached_season_length;
 }
 
+int calendar::turn_of_year() const {
+	return (season_turns() * season) + (turn_number % season_turns());
```

Comment:
The old function contains essentially this logic:
```C++
int turn_of_year() const { 
    return turn_number % DAYS( season_length() * 4 );
}
```

The new function has this:
```C++
int calendar::turn_of_year() const { 
    return DAYS( season_length() ) * season + turn_number % DAYS( season_length() ); 
}
```

According to `calendar::sync`, `season` has either the value `initial_season` (for eternal season) or `turn_number / DAYS( season_length() ) % 4)`.

If the current time is 2 days into the second season (`turn_number = DAYS( season_length() ) + DAYS( 2 )`, `season` be `1` (non-eternal, see `sync`). The old function would return `turn_number`, the new function returns `turn_number + DAYS( season_length() )`.

The function would never return anything `>= DAYS( season_length() * 4 )`, now it can.